core/common/src/rpc/web/WebAppRpcProtocol.ts

```diff
@@ -36,7 +36,7 @@ export interface HttpServerRequest extends Readable {
   statusCode?: number;
   statusMessage?: string;
   socket: any;
-  destroy(error?: Error): void;
+  destroy(error?: Error): any;
```

Comment:
whats the reason for this change?

Reply:
#3947

Reply:
Will we be supporting both node 16 and 18? If so, the return type differs (void on 16 and this on 18).
Do you think I should do it differently?

Reply:
18 returns `this` for destroy? I added an overload:

```ts
  destroy(error?: Error): this;
  destroy(error?: Error): void;
```
But could also use `this|void`? What do you think?

Reply:
I tried this|void, which was a compiler error.
I didn't think of an overload...that seems to work. I will push that.

Reply:
[node@14](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/v14/stream.d.ts#L48)
[node@16](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/v16/stream.d.ts#L411)
[node@18](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/stream.d.ts#L434)

Looks like the return type has been `this` for quite some time now, making me question the need to update to 18 with this pr.

Reply:
> [node@14](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/v14/stream.d.ts#L48) [node@16](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/v16/stream.d.ts#L411) [node@18](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/node/stream.d.ts#L434)
> 
> Looks like the return type has been `this` for quite some time now, making me question the need to update to 18 with this pr.

I guess we're just running an "incorrect" version of @node/types, then (16.11.7)?

I guess my PR here is just getting us to a better version of @node/types and definitely has nothing to do with the broader node18 update, then.

Reply:
> The type used to be `void` back in [node@10](https://unpkg.com/browse/@types/node@10.17.60/stream.d.ts), which we don't currently support anymore anyway.

I will take out my overload for void return then...

Reply:
try changing the types ver from `"16.11.7"` to `"^16.11.7", then run `rush update --full`. Afterwards go back to WebAppRpcProtocol, you might need to restart the TS Server (open the vscode command palette and run TypeScript Restart TS Server). You will see theres a warning about the incorrect return type now

<img width="245" alt="image" src="https://user-images.githubusercontent.com/11051042/190210303-ac40ca3f-44ed-4450-92b2-8e63c3b8a75b.png">

<img width="1002" alt="image" src="https://user-images.githubusercontent.com/11051042/190210469-cb1297ef-71fa-4cc3-b147-9413b8007484.png">

Reply:
if using the carrot doesnt pull the latest 16.x release, then just change it to `16.11.59`

Reply:
Yes, I got the error also. Guess we should have had the caret all along...

Reply:
so what im getting at is we actually dont need to bump to ^18, just bump to ^16 in this pr and change the return type.

@ben-polinsky will work on actually supporting node18, and will include the types/node!18 in his pr

Reply:
> so what im getting at is we actually dont need to bump to ^18, just bump to ^16 in this pr and change the return type.
> 
> @ben-polinsky will work on actually supporting node18, and will include the types/node!18 in his pr

Will do.

Reply:
Done. That hat didn't seem to get rush update to move up, so I hard coded 11.59.